backend/courses/management/commands/form_simple_topics.py

```diff
@@ -0,0 +1,81 @@
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Exists, F, OuterRef
+
+from alert.management.commands.recomputestats import garbage_collect_topics
+from courses.management.commands.load_crosswalk import get_crosswalk_s3
+from courses.models import Course, Topic
+
+
+def form_simple_topics():
+    Course.objects.all().update(topic=None)
+    garbage_collect_topics()
+
+    print("Cleared topics.")
+
+    # create topics for each primary listing
+    primary_listings = Course.objects.filter(primary_listing_id=F("id"))
+    for primary_listing in primary_listings:
+        topic = Topic.objects.create(most_recent=primary_listing)
+        primary_listing.listing_set.all().update(topic=topic)
+
+    print("Created new topics (one per course).")
+
+    # merge topics based on full_code of primary_listing
+    for full_code in primary_listings.values_list("full_code", flat=True):
```

Comment:
Would it be easier to use `collections.defaultdict(set)` based on full_code, then iterate through the dict and merge topics on each individual set?

Could be less iteration if I'm understanding correctly. but don't think it's that big of a difference.

Reply:
I think it would end up being about the same work